js_modules/dagster-ui/packages/ui-core/src/app/ExecutionSessionStorage.tsx

```diff
@@ -31,8 +31,20 @@ export interface PipelineRunTag {
 }
 
 export type SessionBase =
-  | {presetName: string; tags: PipelineRunTag[] | null}
-  | {partitionsSetName: string; partitionName: string | null; tags: PipelineRunTag[] | null};
+  | {
+      presetName: string;
+      tags: PipelineRunTag[] | null;
+    }
+  | {
+      isAssetJob: true; // partition set name not required
+      partitionName: string | null;
+      tags: PipelineRunTag[] | null;
+    }
+  | {
+      partitionsSetName: string; // required for op jobs
+      partitionName: string | null;
+      tags: PipelineRunTag[] | null;
+    };
```

Comment:
Might be nice to give these types individual names just to clarify their purposes.

Reply:
Ahh yeah I like that, will add that instead of the isAssetJob key